WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,25 +46,42 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+$.fn.goodDate = function() {
+  this.parent().siblings('p').removeClass('alert alert-danger');
+  this.removeClass('alert-danger');
+};
+
+$.fn.badDate = function() {
+  this.parent().siblings('p').addClass('alert alert-danger');
+  this.parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
+  this.addClass('alert-danger');
+};
+
 $(function() {
   $('.dropdown-toggle').dropdownHover();
   $('form.are-you-sure').areYouSure();
 
   // Copied (and modified by jfly) from
-  //  https://github.com/zpaulovics/datetimepicker-rails
+  // https://github.com/zpaulovics/datetimepicker-rails
   $('.date_picker.form-control, .datetime_picker.form-control').datetimepicker({
     useStrict: true, keepInvalid: true, useCurrent: false
   });
 
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.error', function(e){
-    $(this).parent().siblings('p').addClass('alert alert-danger');
-    $(this).parent().siblings('p').fadeIn(200).fadeOut(200).fadeIn(200).fadeOut(200).fadeIn(200);
-    $(this).addClass('alert-danger');
-  });
-
-  $('.date_picker.form-control, .datetime_picker.form-control').on('dp.change', function(e){
-    $(this).parent().siblings('p').removeClass('alert alert-danger');
-    $(this).removeClass('alert-danger');
+  $('.date_picker.form-control').on('blur', function(){
+    var date = $(this).val();
+    if(date.match(/^\d{4}-((0\d)|(1[012]))-(([012]\d)|3[01])$/)) {
```

Comment:
Eww. As discussed offline, this won't work for datetime fields. Lets use the bootstrap-datepicker API to get the current format: http://eonasdan.github.io/bootstrap-datetimepicker/Functions/#format, and then use momentjs to check if the format is http://momentjs.com/docs/#/parsing/is-valid/. We should also not be using the blur event to determine when to check this, isn't the `change` event the correct time to check?

I'm thinking something like this:

```javascript
var $this = $(this);
var datetimepicker = $this.data("DateTimePicker");
var val = $this.val();
var valid = moment(val, datetimepicker.format(), true).isValid();
...
```

I also don't see any value in making `goodDate` and `badDate` jquery plugins, this doesn't seem like something we'd use anywhere else.